app/src/main/java/com/github/se/polyfit/data/repository/MealRepository.kt

```diff
@@ -64,7 +64,7 @@ class MealRepository(
   }
 
   suspend fun getAllMeals(): List<Meal> {
-    return mealDao.getAllMeals()
+    return withContext(this.dispatcher) { mealDao.getAllMeals() }
```

Comment:
Was this causing you an error of the type that running this on the main thread might cause  weird behavior ? If so, in the next iteration of the mealRepo, I will update that for all the functions that need to fetch data from the database.

Reply:
It was yeah, thanks!